cloudant-http/src/main/java/com/cloudant/http/interceptors/CookieInterceptor.java

```diff
@@ -245,7 +285,33 @@ private boolean sessionHasStarted(InputStream responseStream) throws IOException
 
     }
 
-    private String extractCookieFromHeaderValue(String cookieHeaderValue) {
-        return cookieHeaderValue.substring(0, cookieHeaderValue.indexOf(";"));
+    private boolean storeCookiesFromResponse(HttpURLConnection connection) {
+
+        // Store any cookies from the response in the CookieManager
+        try {
+            logger.finest("Storing cookie.");
+            cookieManager.put(connection.getURL().toURI(), connection.getHeaderFields());
+            return true;
+        } catch (IOException e) {
+            logger.log(Level.SEVERE, "Failed to read cookie response header", e);
+            return false;
+        } catch (URISyntaxException e) {
+            logger.log(Level.SEVERE, "Failed to convert request URL to URI for cookie storage.");
+            return false;
+        }
+    }
+
+    private String listToSemicolonSeparatedString(List<String> cookieStrings) {
+        // RFC 6265 says multiple cookie pairs should be "; " separated
+        StringBuilder builder = new StringBuilder();
+        int index = 1; // Count from 1 since we will compare to size
+        for (String cookieString : cookieStrings) {
+            builder.append(cookieString);
+            if (index != cookieStrings.size()) {
```

Comment:
Well I was counting from 1 so I didn't, but I did mean to put it at the end of the loop, so good spot. Changed to count from zero and used `index++` in the comparison in 37770a3.